avocado/utils/multipath.py

```diff
@@ -0,0 +1,151 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+
+"""
+Module with multipath related utility functions.
+It needs root access.
+"""
+
+import time
+import logging
+from . import distro
+from . import process
+from . import service
+
+
+def form_conf_mpath_file(blacklist="", defaults_extra=""):
+    """
+    Form a multipath configuration file, and restart multipath service.
+
+    :parm blacklist: Entry in conf file to indicate blacklist section.
+    :parm defaults_extra: Extra entry in conf file in defaults section.
+    """
+    mpath_fp = open("/etc/multipath.conf", "w")
+    mpath_fp.write("defaults {\n")
+    mpath_fp.write("    find_multipaths yes\n")
+    mpath_fp.write("    user_friendly_names yes\n")
+    if defaults_extra:
+        mpath_fp.write("    %s\n" % defaults_extra)
+    mpath_fp.write("}\n")
+    if blacklist:
+        mpath_fp.write("blacklist {\n")
+        mpath_fp.write("    %s\n" % blacklist)
+        mpath_fp.write("}\n")
+    mpath_fp.close()
+    logging.debug(open("/etc/multipath.conf", "r").read())
+    if distro.detect().name == 'Ubuntu':
+        svc_name = "multipath-tools"
+    else:
+        svc_name = "multipathd"
+    time.sleep(5)
+    service.SpecificServiceManager(svc_name).restart()
+
+
+def device_exists(path):
+    """
+    Checks if a given path exists.
+
+    :return: True if path exists, False if does not exist.
+    """
+    cmd = "multipath -l %s" % path
+    if process.system(cmd, ignore_status=True) != 0:
+        return False
+    return True
+
+
+def get_mpath_name(wwid):
+    """
+    Get multipath name for a given wwid.
+
+    :parm wwid: wwid of multipath device.
+
+    :return: Name of multipath device.
+    """
+    if not device_exists(wwid):
+        return
```

Comment:
This empty `return` has an implicit `None`. It's the same thing if you don't have a return. So this function will be more clear if written like this:

```python

def get_mpath_name(wwid):
    if device_exists(wwid):
        cmd = "multipath -l %s" % wwid
        return process.system_output(cmd).split()[0]

```